src/main/java/se/iths/librarysystem/controller/UserRoleController.java

```diff
@@ -0,0 +1,42 @@
+package se.iths.librarysystem.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import se.iths.librarysystem.dto.Role;
+import se.iths.librarysystem.dto.UserWithRole;
+import se.iths.librarysystem.service.UserService;
+import se.iths.librarysystem.validatorservice.RoleValidator;
+import se.iths.librarysystem.validatorservice.UserValidator;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("api/users")
+public class UserRoleController {
+
+    private final UserService userService;
+    private final UserValidator userValidator;
+    private final RoleValidator roleValidator;
+
+    public UserRoleController(UserService userService, UserValidator userValidator, RoleValidator roleValidator) {
+        this.userService = userService;
+        this.userValidator = userValidator;
+        this.roleValidator = roleValidator;
+    }
+
+    @GetMapping("{id}/role")
+    public ResponseEntity<List<Role>> getUserRole(@PathVariable Long id) {
+        List<Role> roles = userService.getUserRoles(id);
```

Comment:
Add userValidator before getting roles?

Reply:
This is handled in the method `getUserRole()`. `userRepository.findById()` returns an optional and therefore the user will always be validated there.  

```
public List<Role> getUserRoles(Long id) {
        UserEntity userEntity = userRepository.findById(id).orElseThrow(() -> new IdNotFoundException("user", id));
        return userEntity.getRoles().stream()
                .map(role -> modelMapper.map(role, Role.class))
                .toList();
    }
```

Reply:
Sweet